version: "3.9"

services:
  python:
    build:
      context: ./build
      dockerfile: Dockerfile
    image: example/python-3.9:fastapi
    container_name: python-3.9
    working_dir: /var/www/app
    restart: "always"
    ports:
      - "58085:8080"
    networks:
      - develop
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://localhost:8080/api/" ]
      interval: 30s
      timeout: 20s
      retries: 3
    command: tail -f /dev/null
    # command: [ "run" ]
    volumes:
      - ../../Python/***REMOVED***:/var/www/app:rw
    environment:
      TZ: "Asia/Shanghai"

  celery_worker:
    image: example/python-3.9:fastapi
    container_name: celery_worker
    working_dir: /var/www/app
    depends_on:
      - "python"
    restart: "no"
    networks:
      - develop
    healthcheck:
      test: [ "CMD", "celery", "-A", "demo", "status" ]
      interval: 30s
      timeout: 20s
      retries: 3
    # command: tail -f /dev/null
    command: celery -A demo worker -l info
    volumes:
      - ../../Python/demo:/var/www/app:rw
    environment:
      TZ: "Asia/Shanghai"
      PYTHONPATH: "/var/www/app"

#  celery_beat:
#    image: example/python-3.9:fastapi
#    container_name: celery_beat
#    working_dir: /var/www/app
#    depends_on:
#      - "python"
#    restart: "no"
#    networks:
#      - develop
#    command: celery -A demo beat -l info
#    volumes:
#      - ../../Python/demo:/var/www/app:rw
#    environment:
#      TZ: "Asia/Shanghai"

networks:
  develop:
    external: true